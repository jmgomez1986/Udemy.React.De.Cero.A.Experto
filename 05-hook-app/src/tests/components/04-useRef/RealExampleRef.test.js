import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Pruebas en <RealExampleRef />', () => {

	const wrapper = shallow( <RealExampleRef />);

	test('debe de mostrarse el componente <RealExampleRef /> correctamente ', () => {

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false); // No deberia existir el componente
		
	});
	
	test('debe de mostrar el componente <MultipleCustomHooks /> ', () => {
		
		wrapper.find('button').simulate('click');

		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true); 
	});
	
});
