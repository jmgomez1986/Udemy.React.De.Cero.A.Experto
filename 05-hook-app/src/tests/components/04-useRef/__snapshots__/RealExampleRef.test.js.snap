// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> debe de mostrarse el componente <RealExampleRef /> correctamente  1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</Fragment>
`;
